lib/Data/Validate/Sanctions.pm

```diff
@@ -57,13 +57,15 @@ sub is_sanctioned {        ## no critic (RequireArgUnpacking)
 
     for my $k (sort keys %$data) {
         foreach my $name (@{$data->{$k}{names}}) {
+            (my $check_name = $name) =~ s/[[:^alpha:]]//g;
+            $check_name = uc($check_name);
             for (@name_variants) {
-                return $k if $name =~ /$_/;
-        }
+                return wantarray ? ($k, $name) : $k if $check_name =~ /$_/;
+            }
         }
     }
 
-    return 0;
+    return wantarray ? (0, '') : 0;
```

Comment:
We need two items - list and name. I can make this function to return single string, like `list:name`. Does it worth it?